idea/src/org/jetbrains/kotlin/idea/inspections/UnnecessaryJavaUsageInspection.kt

```diff
@@ -36,15 +40,18 @@ class UnnecessaryJavaUsageInspection : AbstractKotlinInspection(), CleanupLocalI
 
                 val selectorExpression = expression.selectorExpression ?: return
                 if (selectorExpression !is KtCallExpression) return
-                if (selectorExpression.valueArguments.size != 1) return
-                val value = selectorExpression.valueArguments[0].text
-                val pattern = expression.text.replace(value, "$0")
+                val value = selectorExpression.valueArguments.singleOrNull() ?: return
+
+                val calleeExpression = selectorExpression.calleeExpression as KtSimpleNameExpression
```

Comment:
The calleeExpression is not always a KtSimpleNameExpression, so you should use a safe cast here.